### PR TITLE
Add user engagement as criteria for promoting installability

### PR DESCRIPTION
**Pull request by FormularSumo**
Chromium based browsers (definitely Chrome and Edge, unsure about others) will actively prompt users to install a PWA if it meets the PWA criteria and a user engagement one. I've added some details about this to the making PWAs installable guide.

### CODE REVIEW

**Comment by wbamberg** [comment]
"User engagement" wasn't included in this page because unlike including a manifest, say, it's not a requirement for web developers.

I can see the point of mentioning it, so developers can understand when/why the browser isn't showing an install icon, but think it might be better as an additional note after the note about service workers.

File: files/en-us/web/progressive_web_apps/guides/making_pwas_installable/index.md [hunk 1]
```diff
@@ -73,6 +73,10 @@ For a full description of every member, see the [web app manifest reference docu
 
 For a web app to be installable, it must be served in a [secure context](/en-US/docs/Web/Progressive_web_apps). This usually means that it must be served over HTTPS. Local resources, such as localhost, `127.0.0.1` and `file://` are also considered secure.
 
+### User engagement
+
+Chromium-based browsers such as Chrome and Edge can actively prompt the user to install a PWA if a user engagement criteria has been met. Specifically, having clicked or tapped on the website at least once, and having spent at least 30 seconds viewing the page (at any points in time). See [this guide](https://web.dev/articles/install-criteria) for more information.
```
Comment (wbamberg) [inline files/en-us/web/progressive_web_apps/guides/making_pwas_installable/index.md:78]:
```suggestion
Chromium-based browsers such as Chrome and Edge can will only display an install prompt if the user has clicked or tapped on the website at least once, and has spent at least 30 seconds viewing the page (at any points in time).

See [What does it take to be installable?](https://web.dev/articles/install-criteria) on [web.dev](https://web.dev) for more information.
```

**Comment by FormularSumo** [comment]
That makes sense, I wasn't sure about putting it under requirements either. Having thought about it a bit more, maybe it should go in the installing from the web section, perhaps as a note under the first "when a supporting browser" paragraph? I think that would make sense because it only applies when installing from a browser rather than an app store, and it's not an installability requirement so more relevant to when the PWA is actually being installed.

**Comment by FormularSumo** [comment]
I've moved it into the installing from the web section, and made the distinction between being promoted for installation and being prompted for installation a bit clearer.

**Comment by wbamberg** [comment]
Can you actually reproduce this? I tried, in Chrome, with https://wbamberg.github.io/minimal-pwa, and I get the Install button displayed right way.

File: files/en-us/web/progressive_web_apps/guides/making_pwas_installable/index.md [hunk 2]
```diff
@@ -89,7 +89,7 @@ If you have added your app to the app store, users can install it from there, ju
 
 ## Installation from the web
 
-When a supporting browser determines that a web app meets the installability criteria described earlier, it will promote the app to the user for installation. The user will be offered the chance to install the app. This means you can distribute your PWA as a website, making it discoverable through web search, and also distribute it in app stores, so users can find it there.
+When a supporting browser determines that a web app meets the installability criteria described earlier, it will promote the app to the user for installation by displaying an install button somewhere in the browser UI. Chromium-based browsers such as Chrome and Edge can also display an install prompt if the user has clicked or tapped on the website at least once, and has spent at least 30 seconds viewing the page (at any points in time). See [What does it take to be installable?](https://web.dev/articles/install-criteria) on [web.dev](https://web.dev) for more information. This means you can distribute your PWA as a website, making it discoverable through web search, and also distribute it in app stores, so users can find it there.
```
Comment (hamishwillee) [inline files/en-us/web/progressive_web_apps/guides/making_pwas_installable/index.md:92]: FWIW

> This means you can distribute your PWA as a website, making it discoverable through web search, and also distribute it in app stores, so users can find it there.

The "This means" does  not seem connected to the first part of the sentence - i.e. we've said that user is prompted to install after viewing the page for 30s (and so on) and now they can discover it in web stores and distribute it etc. Where's the connection?

Consider also highlighting the difference with normal sites, that don't promote. If you don't do it here, I'll try in https://github.com/mdn/content/pull/34336

Comment (wbamberg) [reply files/en-us/web/progressive_web_apps/guides/making_pwas_installable/index.md:92]: Yes, I don't like the way this is spliced into the paragraph. But as noted in https://github.com/mdn/content/pull/34335#issuecomment-2187499979, I can't even repro this requirement.

Comment (hamishwillee) [reply files/en-us/web/progressive_web_apps/guides/making_pwas_installable/index.md:92]: You mean the requirement for a 30s wait?

Either way, what needs to happen here is a separate para about how to make something distributable in web stores, and what makes it discoverable (in particular).

Comment (wbamberg) [reply files/en-us/web/progressive_web_apps/guides/making_pwas_installable/index.md:92]: > You mean the requirement for a 30s wait?

Any of it. If I visit  https://wbamberg.github.io/minimal-pwa in a brand-new Chrome, that's never seen the page before, I see the "Install" button right away.

**Comment by FormularSumo** [comment]
Sorry about the confusion, what I'm referring to by an install prompt is a pop-up by the browser which suggests installing an app. I've definitely seen this before, but struggling to reproduce it now. I've been trying to find out more info and apparently Chrome only does this on Android not on desktop. But what it involves is a banner at the bottom of browser which used to say "add to home screen" and I assume now would say "install", and this used to require user engagement, unlike just showing the install button in the search bar on desktop or in the 3 dot menu on Android. You can see an old video of it in action [here](https://youtu.be/9whdWJ4q7jE?si=eTLh-knDBD_Pcs50&t=219). But having visited your minimal-pwa site on Chrome on Android, it seems to fire the beforeinstallprompt event straight away, which should use the same logic. And on other sites I can't now get the Chrome install/add-to-homescreen prompt to show at all, so maybe this is something that has changed since and not been updated in the article. Even on the examples on the [web.dev article](https://web.dev/learn/pwa/installation-prompt/) about it it doesn't seem to do anything for me, just saying "beforeinstallpromptevent supported but not fired yet" (obviously not true as shown in your project).

Going back to the Chrome [blog update](https://developer.chrome.com/blog/update-install-criteria) on PWA installing it said that they were "working to incorporate new signals and display the prompt when the user is most likely to install the app. Developers can still use beforeInstallPrompt() to control the prompt." which would suggest it has now changed, but it's not very clear what to. Maybe Chrome not longer shows these install prompts at all and just relies on the website to fire it. If so, and if it doesn't require any engagement on any platforms which seems to be the case then I think you're right and this issue can be closed. I'll see if I can find time to do some more investigation to double check all this and see if there's been any more recent updates/info.

**Comment by wbamberg** [comment]
Thanks for the clarification, @FormularSumo . I'm going to close this PR since it doesn't seem to apply. We can always rework/reopen if there is something to do here.

**Comment by FormularSumo** [comment]
FWIW I came across an [old web.dev article ](https://web.dev/articles/promote-install)which says:

"Chrome for Android shows a mini infobar to the user, though this can be prevented by calling preventDefault() on the beforeinstallprompt event. If you do not call preventDefault(), the banner will be shown the first time a user visits your site and meets the installability criteria on Android, and then again after approximately 90 days.".

Again, not sure this applies now but interesting that it was spaced apart by 90 days.

**Pull request closed without merging**